include/snowflake/SnowflakeTransferException.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#ifndef SNOWFLAKECLIENT_SNOWFLAKETRANSFEREXCEPTION_HPP
+#define SNOWFLAKECLIENT_SNOWFLAKETRANSFEREXCEPTION_HPP
+
+#include <exception>
+
+namespace Snowflake
+{
+namespace Client
+{
+
+enum TransferError
+{
+  INTERNAL_ERROR,
+  COMPRESSION_ERROR,
+  MKDIR_ERROR,
+  UNSUPPORTED_FEATURE,
+  COLUMN_INDEX_OUT_OF_RANGE,
+  DIR_OPEN_ERROR,
+  COMPRESSION_NOT_SUPPORTED
+};
+
+class SnowflakeTransferException : public std::exception
```

Comment:
This is actually very similar to the base exception class that I have for the C++ wrapper work, which just contains `getCode()` and `what()` functions and private variables to hold the code and msg. I'll pull this change and work it into my branch.